### ISSUE TITLE
JDK 14 & JDK 8 cannot coexist, segmentation fault

### ISSUE THREAD

**Issue by jeff-nevicosi** [AdoptOpenJDK/openjdk-build#1828]
* **Platform and architecture (e.g. Linux/x64**:
Windows
* **Java version (from "java --version") e.g. 1.8.0_181, 11.0.4+11 etc.**:
8.0.252.09 and 14.0.1.7
* **Variant e.g. HotSpot or OpenJ9**:
OpenJ9
* **Have you tested it with other vendors' java**:
Could not reproduce with an old IBM Java 8.0 (build 8.0.5.20 - pwa6480sr5fp20-20180802_01(SR5 FP20)))
* **Steps to reproduce**:
1) Install jdk-14.0.1+7.1_openj9-0.20.0 (Windows x64 - from https://adoptopenjdk.net/releases.html?variant=openjdk14&jvmVariant=openj9)
2) Install jdk8u252-b09.1_openj9-0.20.0 (Windows x64 - from https://adoptopenjdk.net/releases.html?variant=openjdk8&jvmVariant=openj9)
3) Run
"c:\Program Files\AdoptOpenJDK\jdk-8.0.252.09-openj9\bin\java.exe" -version

* **Error messsage and/or description of what is not working as expected**:
Unhandled exception
Type=Segmentation error vmState=0x00000000
Windows_ExceptionCode=c0000139 J9Generic_Signal=00000004 ExceptionAddress=00007FF870DBBF18 ContextFlags=0010000f
Handler1=00007FF82661DE20 Handler2=00007FF829CB8300
RDI=000000000070CF98 RSI=00007FF870DD84C0 RAX=0000000000360034 RBX=00000000C0000139
RCX=00007FF870DDD7B8 RDX=0000000000000000 R8=0000000000000000 R9=0000000000000000
R10=0000000000001112 R11=0000000000001000 R12=0000000000000003 R13=0000000000000003
R14=00007FF8460B8401 R15=0000000000000003
RIP=00007FF870DBBF18 RSP=000000000283C220 RBP=000000000283C8C0 GS=002B
FS=0053 ES=002B DS=002B
XMM0 00007ff870ddd588 (f: 1893586304.000000, d: 6.951752e-310)
XMM1 00007ff870ddd550 (f: 1893586304.000000, d: 6.951752e-310)
XMM2 000000007ffe0385 (f: 2147353472.000000, d: 1.060934e-314)
XMM3 0000000000300035 (f: 3145781.000000, d: 1.554222e-317)
XMM4 00000000007279c0 (f: 7502272.000000, d: 3.706615e-317)
XMM5 000000000000000c (f: 12.000000, d: 5.928788e-323)
XMM6 0000000000000000 (f: 0.000000, d: 0.000000e+000)
XMM7 0000000000000000 (f: 0.000000, d: 0.000000e+000)
XMM8 0000000000000000 (f: 0.000000, d: 0.000000e+000)
XMM9 0000000000000000 (f: 0.000000, d: 0.000000e+000)
XMM10 0000000000000000 (f: 0.000000, d: 0.000000e+000)
XMM11 0000000000000000 (f: 0.000000, d: 0.000000e+000)
XMM12 0000000000000000 (f: 0.000000, d: 0.000000e+000)
XMM13 0000000000000000 (f: 0.000000, d: 0.000000e+000)
XMM14 0000000000000000 (f: 0.000000, d: 0.000000e+000)
XMM15 0000000000000000 (f: 0.000000, d: 0.000000e+000)
Module=C:\WINDOWS\SYSTEM32\ntdll.dll
Module_base_address=00007FF870CC0000 Offset_in_DLL=00000000000fbf18
Target=2_90_20200422_667 (Windows 10 10.0 build 18363)
CPU=amd64 (12 logical CPUs) (0x7ef625000 RAM)
----------- Stack Backtrace -----------
RtlRaiseStatus+0x18 (0x00007FF870DBBF18 [ntdll+0xfbf18])
RtlCharToInteger+0x35c (0x00007FF870D2EEBC [ntdll+0x6eebc])
memset+0x941b (0x00007FF870D6C39B [ntdll+0xac39b])
LdrGetDllHandleByMapping+0x578 (0x00007FF870CEE2A8 [ntdll+0x2e2a8])
LdrGetDllHandleByMapping+0xb8f (0x00007FF870CEE8BF [ntdll+0x2e8bf])
RtlMultiByteToUnicodeSize+0x410 (0x00007FF870CE24C0 [ntdll+0x224c0])
RtlMultiByteToUnicodeSize+0x178 (0x00007FF870CE2228 [ntdll+0x22228])
LdrLoadDll+0xe4 (0x00007FF870CE16E4 [ntdll+0x216e4])
LoadLibraryExW+0x170 (0x00007FF86DDA5630 [KERNELBASE+0x25630])
?si_user_export@@YAXXZ+0xa6f88 (0x00007FF849D81978 [ctiuser+0xb1978])
(0x0000000068018119 [QIPCAP64+0x18119])
?si_user_export@@YAXXZ+0xa74ae (0x00007FF849D81E9E [ctiuser+0xb1e9e])
j9port_init_library+0x2e88f (0x00007FF829CC125F [j9prt29+0x3125f])
J9_CreateJavaVM+0x27144 (0x00007FF8266502F4 [j9vm29+0xb02f4])
J9_CreateJavaVM+0x2689b (0x00007FF82664FA4B [j9vm29+0xafa4b])
J9_CreateJavaVM+0x27519 (0x00007FF8266506C9 [j9vm29+0xb06c9])
JVM_OnLoad+0x28f (0x00007FF826B68ADF [jclse29+0x38adf])
J9VMDllMain+0x131 (0x00007FF826B69351 [jclse29+0x39351])
J9_CreateJavaVM+0x7bd9 (0x00007FF826630D89 [j9vm29+0x90d89])
J9_CreateJavaVM+0x94490 (0x00007FF8266BD640 [j9vm29+0x11d640])
J9_CreateJavaVM+0xd910 (0x00007FF826636AC0 [j9vm29+0x96ac0])
J9_CreateJavaVM+0xf34d (0x00007FF8266384FD [j9vm29+0x984fd])
j9port_init_library+0x26d3f (0x00007FF829CB970F [j9prt29+0x2970f])
J9_CreateJavaVM+0xf5cd (0x00007FF82663877D [j9vm29+0x9877d])
J9_CreateJavaVM+0x89 (0x00007FF826629239 [j9vm29+0x89239])
JVM_Startup+0x1099 (0x00007FF82A571369 [jvm+0x11369])
JNI_CreateJavaVM+0x351 (0x00007FF8615183E1 [jvm+0x83e1])
(0x00007FF70CB5238A [java+0x238a])
(0x00007FF70CB5D007 [java+0xd007])
(0x00007FF70CB5D09B [java+0xd09b])
BaseThreadInitThunk+0x14 (0x00007FF86F387BD4 [KERNEL32+0x17bd4])
RtlUserThreadStart+0x21 (0x00007FF870D2CE51 [ntdll+0x6ce51])
---------------------------------------
JVMDUMP039I Processing dump event "gpf", detail "" at 2020/06/02 13:05:47 - please wait.

--

This can be worked around by removing JDK14 from the PATH, but in the described case, JDK8 is above JDK14 in the path, so it should not cause issues.

**Comment by aahlenst** [comment]
I checked Liberica 8 and Zulu 8. Both have the file system structure with the `jre`:

```
$ find jdk8u252 -name "verify.dll"
jdk8u252/jre/bin/verify.dll
$ find zulu8.46.0.19-ca-jdk8.0.252-win_x64 -name "verify.dll" 
zulu8.46.0.19-ca-jdk8.0.252-win_x64/jre/bin/verify.dll
```

Azul recommends to point `PATH` to `bin/`: https://docs.azul.com/zulu/zuludocs/ZuluUserGuide/PostInstallationTasks/AddPathsToZulu_Windows.htm. BellSoft does the same: https://bell-sw.com/pages/liberica_install_guide-8u252/#installing-standalone-package-on-microsoft-windows.

**Comment by karianna** [comment]
> I checked Liberica 8 and Zulu 8. Both have the file system structure with the `jre`:
> 
> ```
> $ find jdk8u252 -name "verify.dll"
> jdk8u252/jre/bin/verify.dll
> $ find zulu8.46.0.19-ca-jdk8.0.252-win_x64 -name "verify.dll" 
> zulu8.46.0.19-ca-jdk8.0.252-win_x64/jre/bin/verify.dll
> ```
> 
> Azul recommends to point `PATH` to `bin/`: https://docs.azul.com/zulu/zuludocs/ZuluUserGuide/PostInstallationTasks/AddPathsToZulu_Windows.htm. BellSoft does the same: https://bell-sw.com/pages/liberica_install_guide-8u252/#installing-standalone-package-on-microsoft-windows.

@aahlenst Sounds like a workaround is to provide some documentation until we dig a little further?

**Comment by aahlenst** [comment]
I'm not able to reproduce the error. I tried both with HotSpot and OpenJ9 on Windows 10. I installed AdoptOpenJDK 14 first, then 8 on top, both using the MSI and default options (configures `PATH` but leaves `JAVA_HOME` alone).

HotSpot:

```
PS C:\Users\Administrator> echo $Env:PATH
C:\Program Files\AdoptOpenJDK\jdk-8.0.252.09-hotspot\bin;C:\Program Files\AdoptOpenJDK\jdk-14.0.1.7-hotspot\bin;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Windows\System32\OpenSSH\;C:\ProgramData\chocolatey\bin;C:\Program Files\Git\cmd;C:\Program Files (x86)\dotnet\;C:\Program Files\dotnet\;C:\Program Files\Microsoft SQL Server\130\Tools\Binn\;C:\Program Files\Microsoft SQL Server\Client SDK\ODBC\170\Tools\Binn\;C:\Users\Administrator\AppData\Local\Microsoft\WindowsApps;
PS C:\Users\Administrator> java -version
openjdk version "1.8.0_252"
OpenJDK Runtime Environment (AdoptOpenJDK)(build 1.8.0_252-b09)
OpenJDK 64-Bit Server VM (AdoptOpenJDK)(build 25.252-b09, mixed mode)
PS C:\Users\Administrator> & 'C:\Program Files\AdoptOpenJDK\jdk-14.0.1.7-hotspot\bin\java.exe' -version
openjdk version "14.0.1" 2020-04-14
OpenJDK Runtime Environment AdoptOpenJDK (build 14.0.1+7)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 14.0.1+7, mixed mode, sharing)
PS C:\Users\Administrator>
```

OpenJ9:

```
PS C:\Users\Administrator> echo $Env:PATH
C:\Program Files\AdoptOpenJDK\jdk-8.0.252.09-openj9\bin;C:\Program Files\AdoptOpenJDK\jdk-14.0.1.7-openj9\bin;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Windows\System32\OpenSSH\;C:\ProgramData\chocolatey\bin;C:\Program Files\Git\cmd;C:\Program Files (x86)\dotnet\;C:\Program Files\dotnet\;C:\Program Files\Microsoft SQL Server\130\Tools\Binn\;C:\Program Files\Microsoft SQL Server\Client SDK\ODBC\170\Tools\Binn\;C:\Users\Administrator\AppData\Local\Microsoft\WindowsApps;
PS C:\Users\Administrator> java -version
openjdk version "1.8.0_252"
OpenJDK Runtime Environment (build 1.8.0_252-b09)
Eclipse OpenJ9 VM (build openj9-0.20.0, JRE 1.8.0 Windows 10 amd64-64-Bit Compressed References 20200422_667 (JIT enabled, AOT enabled)
OpenJ9   - 05fa2d361
OMR      - d4365f371
JCL      - 5e623848e9 based on jdk8u252-b09)
PS C:\Users\Administrator> echo $Env:JAVA_HOME
PS C:\Users\Administrator> & 'C:\Program Files\AdoptOpenJDK\jdk-14.0.1.7-openj9\bin\java.exe' -version
openjdk version "14.0.1" 2020-04-14
OpenJDK Runtime Environment AdoptOpenJDK (build 14.0.1+7)
Eclipse OpenJ9 VM AdoptOpenJDK (build openj9-0.20.0, JRE 14 Windows 10 amd64-64-Bit Compressed References 20200422_50 (JIT enabled, AOT enabled)
OpenJ9   - 05fa2d361
OMR      - d4365f371
JCL      - 5757187cae based on jdk-14.0.1+7)
PS C:\Users\Administrator> & 'C:\Program Files\AdoptOpenJDK\jdk-8.0.252.09-openj9\bin\java.exe' -version
openjdk version "1.8.0_252"
OpenJDK Runtime Environment (build 1.8.0_252-b09)
Eclipse OpenJ9 VM (build openj9-0.20.0, JRE 1.8.0 Windows 10 amd64-64-Bit Compressed References 20200422_667 (JIT enabled, AOT enabled)
OpenJ9   - 05fa2d361
OMR      - d4365f371
JCL      - 5e623848e9 based on jdk8u252-b09)
PS C:\Users\Administrator>
```

I also tested on a machine without Visual Studio (fresh WIndows image, nothing installed except AdoptOpenJDK):

```
PS C:\Users\Administrator> echo $Env:PATH
C:\Program Files\AdoptOpenJDK\jdk-8.0.252.09-openj9\bin;C:\Program Files\AdoptOpenJDK\jdk-14.0.1.7-openj9\bin;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Windows\System32\OpenSSH\;C:\Users\Administrator\AppData\Local\Microsoft\WindowsApps;
PS C:\Users\Administrator> java -version
openjdk version "1.8.0_252"
OpenJDK Runtime Environment (build 1.8.0_252-b09)
Eclipse OpenJ9 VM (build openj9-0.20.0, JRE 1.8.0 Windows 10 amd64-64-Bit Compressed References 20200422_667 (JIT enabled, AOT enabled)
OpenJ9   - 05fa2d361
OMR      - d4365f371
JCL      - 5e623848e9 based on jdk8u252-b09)
PS C:\Users\Administrator> & 'C:\Program Files\AdoptOpenJDK\jdk-8.0.252.09-openj9\bin\java.exe' -version
openjdk version "1.8.0_252"
OpenJDK Runtime Environment (build 1.8.0_252-b09)
Eclipse OpenJ9 VM (build openj9-0.20.0, JRE 1.8.0 Windows 10 amd64-64-Bit Compressed References 20200422_667 (JIT enabled, AOT enabled)
OpenJ9   - 05fa2d361
OMR      - d4365f371
JCL      - 5e623848e9 based on jdk8u252-b09)
PS C:\Users\Administrator> & 'C:\Program Files\AdoptOpenJDK\jdk-14.0.1.7-openj9\bin\java.exe' -version
openjdk version "14.0.1" 2020-04-14
OpenJDK Runtime Environment AdoptOpenJDK (build 14.0.1+7)
Eclipse OpenJ9 VM AdoptOpenJDK (build openj9-0.20.0, JRE 14 Windows 10 amd64-64-Bit Compressed References 20200422_50 (JIT enabled, AOT enabled)
OpenJ9   - 05fa2d361
OMR      - d4365f371
JCL      - 5757187cae based on jdk-14.0.1+7)
PS C:\Users\Administrator>
```

@jeff-nevicosi Did I miss something?

**Comment by jeff-nevicosi** [comment]
@aahlenst, I can reproduce this from a clean Windows 19 Server install.  Visual Studio was not installed, but some visual frameworks were...  Here is the full list of installed programs:

```
AdoptOpenJDK with OpenJ9 14.0.1.7 (x64)
AdoptOpenJDK with OpenJ9 8.0.252.09 (x64)
Microsoft .NET Framework 4.8 SDK
Microsoft .NET Framework 4.8 Targeting Pack
Microsoft Visual C++ 2008 Redistributable x64 9.0.30729.6161
Microsoft Visual C++ 2008 Redistributable x86 9.0.30729.4148
Symantec Endpoint Protection
VMware Tools
Windows Software Development Kit - Windows 10.0.19041.1
```

Here is the full path:

```
Path=C:\Program Files\AdoptOpenJDK\jdk-8.0.252.09-openj9\bin;C:\Program Files\AdoptOpenJDK\jdk-14.0.1.7-openj9\bin;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Windows\System32\OpenSSH\;C:\Program Files (x86)\Windows Kits\10\Windows Performance Toolkit\;C:\Users\Administrator\AppData\Local\Microsoft\WindowsApps;
```

Any other info you require?

**Comment by jeff-nevicosi** [comment]
@aahlenst, unless there's something else you want me to check, could you remove the "Waiting on OP" label?  I do not have admin access to do this.

**Comment by aahlenst** [comment]
@jeff-nevicosi I tried with Windows Server 2019 1809 and it works, even with .NET Framework 4.8 and Windows 10 SDK. I could not install VMware Tools (my machines are on a different hypervisor) and Symantec Endpoint Protection (I don't have that). 

So, now we need your help: From the list of programs you gave, Symantec Endpoint Protection is the most likely candidate to cause this. Could you please test first on an empty machine if the problem persists? If it does not, could you install one program by one (beginning with Symantec Endpoint Protection) and see if the problem comes back?

**Comment by jeff-nevicosi** [comment]
You were correct.  SEP was the culprit.  The windbg traces look almost identical except for one line. SYSFER.DLL, which controls the Symantec Application & Device Control feature.  I've tried adding a client exception for java.exe, but that was not successful.  I could not add an exception to the SEP manager (as described here:  https://community.broadcom.com/symantecenterprise/communities/community-home/librarydocuments/viewdocument?DocumentKey=f193716a-8a25-4e4e-abbd-f7c0eac29964&CommunityKey=1ecf5f55-9545-44d6-b0f4-4e4a7f5f5e68&tab=librarydocuments or  https://knowledge.broadcom.com/external/article/180778/how-to-create-an-application-exception-i.html) due to company profile restrictions.

With SEP installed  (fails):

```
CommandLine: "C:\Program Files\AdoptOpenJDK\jdk-8.0.252.09-openj9\bin\java.exe"

************* Path validation summary **************
Response                         Time (ms)     Location
Deferred                                       srv*
Symbol search path is: srv*
Executable search path is: 
ModLoad: 00007ff7`d2560000 00007ff7`d2595000   java.exe
ModLoad: 00007ffa`8dab0000 00007ffa`8dc9d000   ntdll.dll
ModLoad: 00007ffa`8c940000 00007ffa`8c9f3000   C:\Windows\System32\KERNEL32.DLL
ModLoad: 00007ffa`8a2d0000 00007ffa`8a563000   C:\Windows\System32\KERNELBASE.dll
**ModLoad: 00000000`5db90000 00000000`5dc23000   C:\Windows\System32\SYSFER.DLL**
ModLoad: 00007ffa`8c890000 00007ffa`8c933000   C:\Windows\System32\ADVAPI32.dll
ModLoad: 00007ffa`8cd30000 00007ffa`8cdce000   C:\Windows\System32\msvcrt.dll
ModLoad: 00007ffa`8d540000 00007ffa`8d5de000   C:\Windows\System32\sechost.dll
ModLoad: 00007ffa`8c630000 00007ffa`8c753000   C:\Windows\System32\RPCRT4.dll
ModLoad: 00007ffa`8cb60000 00007ffa`8ccf7000   C:\Windows\System32\USER32.dll
ModLoad: 00007ffa`8ac60000 00007ffa`8ac80000   C:\Windows\System32\win32u.dll
ModLoad: 00007ffa`8d980000 00007ffa`8d9a9000   C:\Windows\System32\GDI32.dll
ModLoad: 00007ffa`8a800000 00007ffa`8a99a000   C:\Windows\System32\gdi32full.dll
ModLoad: 00007ffa`8aa20000 00007ffa`8aac0000   C:\Windows\System32\msvcp_win.dll
ModLoad: 00007ffa`8aac0000 00007ffa`8abbc000   C:\Windows\System32\ucrtbase.dll
ModLoad: 00007ffa`78d10000 00007ffa`78f89000   C:\Windows\WinSxS\amd64_microsoft.windows.common-controls_6595b64144ccf1df_6.0.17763.107_none_05b63884071e656e\COMCTL32.dll
ModLoad: 00007ffa`8d5e0000 00007ffa`8d90e000   C:\Windows\System32\combase.dll
ModLoad: 00007ffa`8a9a0000 00007ffa`8aa1e000   C:\Windows\System32\bcryptPrimitives.dll
(9bc.10b8): Break instruction exception - code 80000003 (first chance)
...
```

Without SEP (success):

```
CommandLine: "C:\Program Files\AdoptOpenJDK\jdk-8.0.252.09-openj9\bin\java.exe"

************* Path validation summary **************
Response                         Time (ms)     Location
Deferred                                       srv*
Symbol search path is: srv*
Executable search path is: 
ModLoad: 00007ff6`4acf0000 00007ff6`4ad25000   java.exe
ModLoad: 00007ff8`e9ac0000 00007ff8`e9cad000   ntdll.dll
ModLoad: 00007ff8`e6e80000 00007ff8`e6f33000   C:\Windows\System32\KERNEL32.DLL
ModLoad: 00007ff8`e5f80000 00007ff8`e6213000   C:\Windows\System32\KERNELBASE.dll
ModLoad: 00007ff8`e7f80000 00007ff8`e8023000   C:\Windows\System32\ADVAPI32.dll
ModLoad: 00007ff8`e8040000 00007ff8`e80de000   C:\Windows\System32\msvcrt.dll
ModLoad: 00007ff8`e7df0000 00007ff8`e7e8e000   C:\Windows\System32\sechost.dll
ModLoad: 00007ff8`e7610000 00007ff8`e7733000   C:\Windows\System32\RPCRT4.dll
ModLoad: 00007ff8`e80e0000 00007ff8`e8277000   C:\Windows\System32\USER32.dll
ModLoad: 00007ff8`e5f60000 00007ff8`e5f80000   C:\Windows\System32\win32u.dll
ModLoad: 00007ff8`e8280000 00007ff8`e82a9000   C:\Windows\System32\GDI32.dll
ModLoad: 00007ff8`e5be0000 00007ff8`e5d7a000   C:\Windows\System32\gdi32full.dll
ModLoad: 00007ff8`e6970000 00007ff8`e6a10000   C:\Windows\System32\msvcp_win.dll
ModLoad: 00007ff8`e6a10000 00007ff8`e6b0c000   C:\Windows\System32\ucrtbase.dll
ModLoad: 00007ff8`d5140000 00007ff8`d53b9000   C:\Windows\WinSxS\amd64_microsoft.windows.common-controls_6595b64144ccf1df_6.0.17763.107_none_05b63884071e656e\COMCTL32.dll
ModLoad: 00007ff8`e6ff0000 00007ff8`e731e000   C:\Windows\System32\combase.dll
ModLoad: 00007ff8`e6c50000 00007ff8`e6cce000   C:\Windows\System32\bcryptPrimitives.dll
(157c.1674): Break instruction exception - code 80000003 (first chance)
...
```

**Comment by aahlenst** [comment]
Good to hear. Might be a good idea to report this to Broadcom if you can do that.
